app/src/backend/formulas.ts

```diff
@@ -321,8 +408,14 @@ export function handleVkt1Formula(
 
 export function handleMethaneCommitmentFormula(
   activityValue: ActivityValue,
+  inventoryValue: InventoryValue,
 ): Gas[] {
-  const data = activityValue.activityData;
+  const data = convertDataToDefaultUnit(
+    // use convert all the values
+    activityValue,
+    inventoryValue.inputMethodology as string,
```

Comment:
Instead of converting the type it might be good to handle null cases in some place beforehand?

Reply:
the input methodology will always be required for conversions because we need to access the methodology in the manual input json file

Reply:
Yes, I meant doing a null check and throwing/ logging an error here instead of effectively using the `!` Typescript postfix operator by using `as string`

Reply:
Checking it and returning/ throwing in that case means it is no longer considered nullable in the current block by the TS compiler, meaning you can leave out `as string` in all of those places.

Reply:
noted and corrected. THanks